convert2rhel/backup/packages.py

```diff
@@ -136,10 +136,10 @@ def enable(self):
         loggerinst.info("Backing up the packages: %s." % ",".join(self.pkgs))
         loggerinst.debug("Using repository files stored in %s." % self.reposdir)
 
-        if self.reposdir:
+        if self.reposdir and os.path.exists(self.reposdir):
             # If nothing is inside the directory, or it does not exist, let's
             # just not use it to download the packages.
-            if len(os.listdir(self.reposdir)) == 0 or not os.path.exists(self.reposdir):
+            if len(os.listdir(self.reposdir)) == 0:
```

Comment:
We could combine all three but I feel like the this is the best solution atm

You could have done all three as if-statements go left to right (usually)

E.g. `if not os.path.exists(self.reposdir) or len(os.listdir(self.reposdir)) == 0:` would also have fixed the issue

However, please update the comments too! :pray:

Reply:
Actually scratch that.. it's intended that we go to the log no matter what, so the exception is false but implementation was (mostly) correct as it was before

However, we should still change it to be something like

```
if reposdir
	if (reposdir exists and empty) or reposdir does not exist:
		log and set reposdir empty
```

Reply:
I don't think it will work like that... If the path doesn't exist and we pass that to os.listdir(), it will cause an exception as the path doesn't exist. Let me try that locally

Reply:
OOoh, that's funny! 

So, I didn't know that using `and` here would not cause an exception. If we use `or`, then the exception could arise, with `and` it doesn't. That's a new learning. 

I will switch the code as you proposed.

Reply:
Updated!

Reply:
> So, I didn't know that using `and` here would not cause an exception. If we use `or`, then the exception could arise, with `and` it doesn't. That's a new learning.

It has to do with if-statements evaluating left to right, so if you have `if os.path.exists(path) and path` it would fail. You can also do optimization with this in mind in general. Such as finding something from a database and evaluating it. Happens a lot more commonly in other OOP languages.